Amanda/1173 signin refresh

AmandaRiu opened the pull request:
This PR fixes #1173 by completing the following:
- Adding a "refresh app" link to the error message the users sees when presented with the "not a woo store" error.
- Reordered the primary and secondary buttons to match the design specs

<img src="https://user-images.githubusercontent.com/5810477/59865603-aa8ef480-9346-11e9-8a9a-45119c6708b2.gif" width="300"/>

## To Test
1. Log in with a site that has Jetpack installed, but not woo
2. Get to the "not a woo store" error screen
3. Install and activate the Woo plugin on the site
4. Click "refresh app" in the error message
5. Confirm the app refreshes and logs the user in successfully

Update release notes:

- [x] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

Commit:
Add new tracks event

Commit:
Update string for not woo store message

Split string into two parts for use with spannables since only
part of the text will have a different color and will be clickable.

Commit:
Format partial not woo store message color using spannable

Commit:
Use a custom ClickableSpan to properly format and handle click

Commit:
Add track event to track when the user clicked to refresh app

Commit:
Refresh app by fetching site from api and checking for woocommerce

Commit:
Reorder the primary and secondary buttons on the site picker

Commit:
Update release notes to include sign in changes

anitaa1990 reviewed (changes requested):
@AmandaRiu the code changes look really good! The only minor issue I found is that the `Try Another account` button is visible when trying to switch stores. 

<img width="300" src="https://user-images.githubusercontent.com/22608780/60407561-e6dd1300-9bd8-11e9-9f51-a4f7d74d9157.png">

- I also found a scenario where I did not installed WooCommerce and activated it. But I did not run the setup wizard. While the current logic works good and I am redirected to the stats page, I get `Error fetching orders` since the WooCommerce setup is not yet run, I guess. This is not related to this PR and I'm not sure if this is something we would be able to find out from the API but just thought I would mention it here in case we are able to handle it
<img width="300" src="https://user-images.githubusercontent.com/22608780/60407583-fbb9a680-9bd8-11e9-83a1-e6f89f0e42e8.png">

Commit:
Give buttons generic names since the labels will change dynamically

The primary button is the purple button with white text.
The secondary button is a white button with purple text for
secondary actions if available. Primary button should always be visible,
secondary button is hidden until needed.

Commit:
Update secondary button label to "View Connected Stores"

"Continue" is a primary action so it will share the primary button
along with "try another account"

Commit:
Remove unnecessary logic to show the primary button, it's always visible

Also cleaned up setup logic that was redundant.

Commit:
Hide "Connected Stores" button when store list is displayed

Commit:
Updated to use the appropriate primary/secondary button as needed

AmandaRiu commented:
@anitaa1990 Thank you for the review and awesome catch! With the new designs the primary button label is changed dynamically depending on the current state of the activity. The naming of these buttons were causing me confusion while troubleshooting them so I decided to rename them to simplify things in d1e002182252c19810a6c235cf6b2497cba7247f:
- `button_primary`: this is the purple button with white text and should always be visible
- `button_secondary`: this is the white button with purple text and will be hidden until needed. 

Fixes:
- updated the secondary button to always be "View connected stores" in feee4250f66854624a46cf3ebf6cda148f60af4f
- hide the secondary button when the store list is displayed 0a587ea7367069d4c4d7f94b882351f9c754d064
- fix logic to use the appropriate primary/secondary buttons as needed 953c180c9552393a310827db7242ddfcf206e84c

Ready for another round 👍

AmandaRiu commented:
> I also found a scenario where I did not installed WooCommerce and activated it. But I did not run the setup wizard. While the current logic works good and I am redirected to the stats page, I get Error fetching orders since the WooCommerce setup is not yet run, I guess. This is not related to this PR and I'm not sure if this is something we would be able to find out from the API but just thought I would mention it here in case we are able to handle it

This is how the old logic works as well. You can always select a store that isn't properly configured. I think we'll need a research spike to determine if it's possible to know if a store isn't configured.

anitaa1990 reviewed (approved):
Looks great! LGTM :shipit:

anitaa1990 commented:
> This is how the old logic works as well. You can always select a store that isn't properly configured. I think we'll need a research spike to determine if it's possible to know if a store isn't configured.

Thanks @AmandaRiu ! I've created a separate issue for this [here](https://github.com/woocommerce/woocommerce-android/issues/1195) and also validated the new event added from this PR.
